### PR TITLE
Log viewport size, per-view counts, and indirect discovery (SCP-5183)

### PR DESCRIPTION
**Pull request by eweitz**
This enables staff to answer relevant questions about product usage and user experience, e.g.:

### How big is the user's window?
New `viewportWidth` and `viewportHeight` properties report the size of the user's app window.  This complements `screen_width` and `screen_height`, which report the size of the user's screen / device / monitor.  (The casing of these properties differs because the latter pair is "built in" as part of the mirrored Mixpanel suite, whereas the former is not.)

Knowing this can help inform how much effort we invest tailoring layouts for various viewport dimensions.

### How many times has this thing been clicked in this page view?
We sometimes make changes aimed at boosting general engagement.  While we can assess event-level total counts or statistics per user, we haven't been able to assess how engagement _within a given page view_ changes.  

The new properties `numEventsThisPageLoad:thisName` and `numEventsSincePageLoad:thisText` enable that at a reasonably granular level, by reporting the number of times an event with "this" name (i.e., the name of the event logged to Mixpanel), and "this" displayed text are done in the current page view.  The `numEventsSincePageLoad:clicks` does similarly, but for a higher-level metric -- the number of logged clicks in the given page view.

Knowing this can help assess the per-view impact of UX changes on engagement, both overall and for specific elements.

### Did the user see the gene they searched in a gene discovery UI component?
Sometimes, users learn about a gene through the DE table or ideogram, but don't search directly through those widgets as we'd typically expect.  The `search` event has new properties -- `numSearchedGenesInDifferentialExpressionTable` and `numSearchedGenesInIdeogram` -- which will help us systematically measure how often that happens.

Knowing this can help more accurately assess how much of our search activity is attributable to DE table and ideogram, and whether users understand they can interact through those widgets.

### Test
To manually test:
* Go to a page
* Resize browser
* Open Network panel in DevTools, filter to `bard method:POST`
* Click a button
* In Network panel, find post request to Bard and:
  * Confirm `viewportWidth` and `viewportHeight` appear, and differ from `screen_width` and `screen_height`
  * Confirm `numEventsThisPageLoad:thisName` and `numEventsSincePageLoad:clicks` report 1, and (assuming the element has text or a label) `numEventsSincePageLoad:thisText` reports 1
* Click a _different_ button, and confirm `numEventsThisPageLoad:thisName` and `numEventsSincePageLoad:clicks` report 2, and (assuming the element has text or a label) `numEventsSincePageLoad:thisText` reports 1 _and not 2_
* Go to a study with DE results
* Click "Differential expression" button
* In the main gene search box at upper left in page, type in a gene shown in the DE table, and execute a search for it
* In Bard post, confirm `numSearchedGenesInDifferentialExpressionTable` reports 1
* Click arrow to right of search box to "Return to cluster view"
* In main gene search box, search "PTEN"
* Confirm "PIK3CA" is shown in related genes ideogram
* In main gene search box, search "PIK3CA"
* In Bard post, confirm `numSearchedGenesInIdeogram` reports 1

For richer context, see [SCP demo discussion](https://drive.google.com/file/d/1rdtpUoNjmm-bAsAs3BlAo0Me7XYnJi7Q/view?t=10).  This was a Developer's Choice Day project, tracked in SCP-5183.

### CODE REVIEW

File: app/javascript/components/explore/DifferentialExpressionPanel.jsx [hunk 1]
```diff
@@ -158,6 +158,7 @@ const columnHelper = createColumnHelper()
 function searchGenesFromTable(selectedGenes, searchGenes, logProps) {
   searchGenes(selectedGenes)
 
+  console.log('selectedGenes', selectedGenes)
```
Comment (bistline) [inline app/javascript/components/explore/DifferentialExpressionPanel.jsx:161]: Debug statement

File: app/javascript/lib/metrics-api.js [hunk 2]
```diff
@@ -425,8 +429,38 @@ export function log(name, props = {}) {
     props['tab'] = tab
   }
 
+  props['viewportWidth'] = window.innerWidth
+  props['viewportHeight'] = window.innerHeight
+
   props['timeSincePageLoad'] = Math.round(performance.now())
 
+  // Enables answering e.g.:
```
Comment (bistline) [inline app/javascript/lib/metrics-api.js:437]: Helpful!

**Review by bistline — APPROVED** [review]
Looks good, modulo one debug `console.log` to remove.

**Review by ehanna4 — APPROVED** [review]
Nice!

**Pull request merged**